### DISCUSSION TITLE
Deno/Fresh/Tailwind integration with DaisyUI

### DISCUSSION THREAD

**Discussion by cknight** [saadeghi/daisyui#2629] [Q&A] [answered]
I am using the latest version of Deno Fresh (v1.6) which now supports Tailwind proper.  I've got this all working with the CDN version of DaisyUI, but am wondering how to integrate DaisyUI as a plugin instead as adding any config to the tailwind.config.ts seems to have no effect.

Any help on better integration between Deno/Fresh and DaisyUI would be most appreciated!

**Comment by saadeghi** [comment, marked as answer]
Hey.
I haven't tried Deno, but a `tailwind.config.ts` on an ESM project can be like this:
```js
import daisyui from 'daisyui';

export default {
  content: ['./src/**/*.{js,ts}'],
  plugins: [daisyui],
};
```

https://stackblitz.com/edit/daisyui-vite-6udkoc?file=tailwind.config.ts

Let me know if you have a question.

---

Edit: it's fixed now since `5.0.24` ✅

**Reply by cknight** [reply]
Awesome thanks very much!  For other Fresh visitors, here's how I got it working.  Not super fond of the `as any` cast, but will see if I can figure out types later.

First add the daisyui to your `deno.json` import map.  E.g.
```ts
{
  ...
  "imports": {
    "daisyui": "npm:daisyui@4.4.19",
    ...
  },
  ...
}

```

Then install the plugin as follows in your `tailwind.config.ts`:
```ts
import { type Config } from "tailwindcss";
import daisyui from "daisyui";

export default {
  content: [
    "{routes,islands,components}/**/*.{ts,tsx}",
  ],
  plugins: [daisyui as any],
  daisyui: { 
    themes: [
      "cupcake",
      "dark",
      "dim",
    ],
  },
} as Config;
```

**Comment by MrPaolinsky** [comment]
Update for 2025 and Deno 2 release, it works well.
```ts
import { type Config, type PluginCreator } from "tailwindcss";
import daisyui from "daisyui";

export default {
  content: [
    "{routes,islands,components}/**/*.{ts,tsx,js,jsx}",
  ],
  plugins: [daisyui as PluginCreator],
  daisyui: {
      themes: ["cupcake"]
  }
} satisfies Config;
```

Use ``deno add daisyui`` to install the latest version of daisyui

**Reply by saadeghi** [reply]
Nice!  
Is it Tailwind CSS 3 or 4?

**Reply by MrPaolinsky** [reply]
Im using ``Fresh`` the fullstack framework, and the version of tailwind it installs by default it's 3, I'm guessing it should work with 4 as well.

**Reply by saadeghi** [reply]
Thanks. If you found a guide or a setup to make it work with Tailwind CSS 4 + daisyUI 5 beta let me know so I can add it to https://v5.daisyui.com/docs/install/

**Reply by yesmar12** [reply]
I'm getting an error when I attempt to import daisy ui in the config file. 

```
error: Uncaught (in promise) TypeError: Attempted to load JSON module without specifying "type": "json" attribute in the import statement.
  const tailwindConfig = (await import(url)).default as Config;
                          ^
    at async Module.initTailwind (https://deno.land/x/fresh@1.7.3/plugins/tailwind/compiler.ts:48:27)
    at async initTailwind (https://deno.land/x/fresh@1.7.3/plugins/tailwind.ts:11:10)
    at async Object.configResolved (https://deno.land/x/fresh@1.7.3/plugins/tailwind.ts:91:21)
    at async Promise.all (index 0)
    at async getServerContext (https://deno.land/x/fresh@1.7.3/src/server/context.ts:79:5)
    at async dev (https://deno.land/x/fresh@1.7.3/src/dev/dev_command.ts:57:17)
    at async temp/dev.ts:8:1
```

Minimal solution can be recreated with 
1. `deno run -A -r https://fresh.deno.dev`
2. `deno i -D npm:daisyui@latest`
3. Updating tailwind.config.ts as above. 

Has anyone run into this?

**Reply by yesmar12** [reply]
I have gotten it to work with daisy ui version 4. For now this works for me.

**Reply by saadeghi** [reply]
> I have gotten it to work with daisy ui version 4. For now this works for me.

Yes, it's fixed since `5.0.24` 🎉
